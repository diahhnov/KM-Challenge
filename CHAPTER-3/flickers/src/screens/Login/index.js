import {
  Text,
  View,
  TouchableOpacity,
  ImageBackground,
  ActivityIndicator,
} from 'react-native';
import React, {useState} from 'react';
import {Input, Button} from 'react-native-elements';
import axios from 'axios';
import {BaseUrlApi} from '../../helpers/ApiAcess';

export default function Login({navigation}) {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [loading, setLoding] = useState(false);
  const postLogin = async () => {
    try {
      setLoding(true);
      const body = {
        username: username, // mor_2314
        password: password, // 83r5^_
      };

      const res = await axios.post(`${BaseUrlApi}/auth/login`, body, {
        validateStatus: status => status < 501,
      });
      console.log(res.status);
      if (res.status <= 201) {
        navigation.navigate('Home');
      } else {
        return alert('ERROR!!!, wrong username or password');
      }
    } catch (error) {
      console.log(error);
    } finally {
      setLoding(false);
    }
  };

  if (loading) {
    return (
      <View
        style={{
          flex: 1,
          justifyContent: 'center',
          alignItems: 'center',
          backgroundColor: '#36485f',
        }}>
        <ActivityIndicator />
        <Text style={{color: '#FFF', fontWeight: 'bold'}}>loading</Text>
      </View>
    );
  }

  return (
    <View
      style={{
        alignContent: 'center',
        justifyContent: 'center',
        flex: 1,
        backgroundColor: '#0000',
      }}>
      <ImageBackground
        source={require('../../assets/bioskop.jpg')}
        style={{flex: 1, resizeMode: 'cover', justifyContent: 'center'}}>
        <View style={{marginTop: -150}}>
          <Text
            style={{
              margin: 20,
              textAlign: 'center',
              fontSize: 50,
              fontWeight: 'bold',
              color: '#5D8BF4',
            }}>
            Sign In
          </Text>
          <View
            style={{
              margin: 10,
              alignContent: 'center',
              justifyContent: 'center',
            }}>
            <Input
              style={{height: 50, color: '#5D8BF4'}}
              label="Username"
              placeholder=" "
              onChangeText={text => setUsername(text)}
            />
            <Input
              style={{height: 50, color: '#5D8BF4'}}
              label="Password"
              placeholder=" "
              onChangeText={text => setPassword(text)}
              secureTextEntry={true}
            />
          </View>
          <View
            style={{
              alignItems: 'center',
              justifyContent: 'center',
            }}>
            <Button
              buttonStyle={{borderWidth: 2, borderRadius: 20}}
              containerStyle={{
                width: 200,
              }}
              onPress={postLogin}
              title={'Sign in'}
              titleStyle={{fontWeight: 'bold'}}
            />
            <TouchableOpacity
              style={{marginVertical: 10}}
              onPress={() => navigation.navigate('Register')}>
              <Text style={{color: 'white'}}>SignUp</Text>
            </TouchableOpacity>
          </View>
        </View>
      </ImageBackground>
    </View>
  );
}
