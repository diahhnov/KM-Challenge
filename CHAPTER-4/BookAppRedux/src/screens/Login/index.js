import {
  View,
  Alert,
  StatusBar,
  TextInput,
  TouchableOpacity,
  StyleSheet,
  Image,
} from 'react-native';
import React, {useState} from 'react';
import {useNavigation, useIsFocused} from '@react-navigation/native';
import styles from './style';
import {Login_Api} from '../../helpers/Api';
import axios from 'axios';
import Montserrat from '../../components/Montserrat';

const Login = () => {
  const navigation = useNavigation();
  // const focussed = useIsFocused();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  // const [emailFocussed, setEmailFocussed] = useState(false);
  // const [passwordFocussed, setPasswordFocussed] = useState(false);

  // const checkers = () => {
  //   const regexEmail = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
  //   const emailStat = regexEmail.test(email);

  //   const regexPassword = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])[0-9a-zA-Z]{8,}$/;
  //   const passwordStat = regexPassword.test(email);

  //   if (email.length === 0 && password.length === 0) {
  //     Alert.alert('ERROR', "this can't be empty");
  //   } else {
  //     if (emailStat && passwordStat === true) {
  //       // navigation.navigate('MainRouter');
  //     } else {
  //       Alert.alert('ERROR');
  //     }
  //   }
  // };

  const postLogin = async () => {
    try {
      const body = {
        email: email, // dodidido2022@gmail.com
        password: password, // Dodidido2022.
      };

      const res = await axios.post(`${Login_Api}`, body, {
        validateStatus: status => status < 501,
      });
      console.log(res.status);
      if (res.status <= 201) {
        // navigation.navigate('Home');
      } else {
        return alert('ERROR!!!, wrong username or password');
      }
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <View style={styles.wrapper}>
      <StatusBar
        backgroundColor="white"
        barStyle="dark-content"
        // status={focussed}
      />
      <Image
        source={require('../../assets/image/Buku.png')}
        style={styles.logo}
      />
      <View style={styles.containerInput}>
        <View style={styles.containnerLoginText}>
          <Montserrat style={styles.LoginText}>Login</Montserrat>
        </View>
        <View
        // style={styles.emailInput(emailFocussed)}
        >
          <TextInput
            style={styles.input}
            placeholder="Enter Email"
            onChangeText={text => setEmail(text)}
            // onFocus={setEmailFocussed(true)}
            // onBlur={setEmailFocussed(false)}
          />
        </View>
        <View
        // style={styles.passwordInput(passwordFocussed)}
        >
          <TextInput
            style={styles.input}
            placeholder="Enter Password"
            onChangeText={text => setPassword(text)}
            // onFocus={setPasswordFocussed(true)}
            // onBlur={setPasswordFocussed(false)}
            // secureTextEntry={true}
          />
        </View>
        <View>
          <TouchableOpacity onPress={postLogin} style={styles.buttonLogin}>
            <Montserrat style={styles.textbuttonLogin}>LOGIN</Montserrat>
          </TouchableOpacity>
          <Montserrat style={styles.ask}>Don't have an account</Montserrat>
          <TouchableOpacity onPress={() => navigation.navigate('Register')}>
            <Montserrat style={styles.Register}>REGISTER</Montserrat>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

export default Login;
