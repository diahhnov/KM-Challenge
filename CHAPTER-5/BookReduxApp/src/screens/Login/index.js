import {
  View,
  Alert,
  StatusBar,
  TextInput,
  TouchableOpacity,
  Image,
  RefreshControl,
} from 'react-native';
import React, {useEffect, useState} from 'react';
import axios from 'axios';
import {useDispatch, useSelector} from 'react-redux';
import {ActivityIndicator} from 'react-native-paper';
import AsyncStorage from 'react-native';
import styles from './style';
import {Login_Api} from '../../helpers/Api';
import Montserrat from '../../components/Montserrat';
import {setLoading} from '../utils/globalAction';
import {setToken} from './utils/action';
import {navigate} from '../../helpers/navigate';

function Login({navigation}) {
  const dispatch = useDispatch();
  const {token} = useSelector(state => state.login);
  const {loading} = useSelector(state => state.global);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  /* eslint-disable no-useless-escape */
  /* eslint-comments/no-unlimited-disable */
  const regexEmail = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;

  const postLogin = async () => {
    if (email.length === 0) {
      Alert.alert('Error', 'Email tidak boleh kosong!');
    } else if (password === 0) {
      Alert.alert('Error', 'Password tidak boleh kosong!');
    } else if (!email.match(regexEmail)) {
      Alert.alert('Error, email tidak valid');
    } else {
      try {
        dispatch(setLoading(true));
        const body = {
          email, // dodidido2022@gmail.com
          password, // Dodidido2022.
        };

        const res = await axios.post(`${Login_Api}`, body, {
          validateStatus: status => status < 501,
        });
        console.log(res);
        console.log(res.status);
        if (res.status <= 201) {
          dispatch(setToken(res.data.tokens.access.token));
          navigate('MainRoot');
        } else {
          return Alert.alert('ERROR!!!, wrong username or password');
        }
      } catch (error) {
        console.log(error);
      } finally {
        dispatch(setLoading(false));
      }
    }
  };

  const [refresh, setRefresh] = useState(false);
  const onRefreshing = () => {
    setRefresh(true);
    setTimeout(() => {
      setRefresh(false);
    }, 3000);
  };
  /* eslint no-shadow: ["error", { "hoist": "functions" }] */
  /* eslint-env es6 */
  useEffect(() => {
    const tokenCheck = async () => {
      try {
        const res = await AsyncStorage.getItem('@token');
        if (res !== null) {
          dispatch(setToken(res));
          navigation.navigate('MainRoot');
        }
      } catch (error) {
        console.log(error);
      }
    };
    tokenCheck();
  }, [token, dispatch, navigation]);

  return (
    <View
      refreshControl={
        <RefreshControl refreshing={refresh} onRefresh={onRefreshing} />
      }
      style={styles.wrapper}>
      <StatusBar backgroundColor="white" barStyle="dark-content" />
      <Image
        source={require('../../assets/image/Buku.png')}
        style={styles.logo}
      />
      <View style={styles.containerInput}>
        <View style={styles.containnerLoginText}>
          <Montserrat style={styles.LoginText}>Login</Montserrat>
        </View>
        <View>
          <TextInput
            style={styles.input}
            placeholder="Enter Email"
            onChangeText={text => setEmail(text)}
          />
        </View>
        <View>
          <TextInput
            style={styles.input}
            placeholder="Enter Password"
            onChangeText={text => setPassword(text)}
          />
        </View>
        <View>
          {loading ? (
            <ActivityIndicator />
          ) : (
            <TouchableOpacity
              onPress={postLogin}
              style={styles.buttonLogin}
              testID="loginButton">
              <Montserrat style={styles.textbuttonLogin}>LOGIN</Montserrat>
            </TouchableOpacity>
          )}
          <Montserrat style={styles.ask}>Don't have an account</Montserrat>
          <TouchableOpacity onPress={() => navigation.navigate('Register')}>
            <Montserrat style={styles.Register}>REGISTER</Montserrat>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}

export default Login;
