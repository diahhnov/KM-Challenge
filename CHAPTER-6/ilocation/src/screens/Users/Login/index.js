import {
  View,
  Alert,
  StatusBar,
  TextInput,
  TouchableOpacity,
  RefreshControl,
} from 'react-native';
import React, {useEffect, useState} from 'react';
import axios from 'axios';
import {useDispatch, useSelector} from 'react-redux';
import {ActivityIndicator} from 'react-native-paper';
import AsyncStorage from 'react-native';
import styles from './style';
import {Login_Api} from '../../../helpers/Api';
import Montserrat from '../../../helpers/Montserrat';
import {setLoading} from '../../utils/globalAction';
import {setToken} from './utils/action';
import {navigate} from '../../../helpers/navigate';
import BioAuth from '../BioAuth';
import {
  GoogleSignin,
  statusCodes,
  GoogleSigninButton,
} from '@react-native-google-signin/google-signin';
import {LoginButton, AccessToken, LoginManager} from 'react-native-fbsdk-next';
import {moderateScale} from 'react-native-size-matters';

function Login({navigation}) {
  const dispatch = useDispatch();
  const {token} = useSelector(state => state.login);
  const {loading} = useSelector(state => state.global);
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  /* eslint-disable no-useless-escape */
  /* eslint-comments/no-unlimited-disable */
  const regexEmail = /^[a-zA-z0-9._-]+@[a-zA-z0-9]+\.[a-zA-]/;

  const postLogin = async () => {
    if (email.length === 0) {
      Alert.alert('Error', 'Email tidak boleh kosong!');
    } else if (password === 0) {
      Alert.alert('Error', 'Password tidak boleh kosong!');
    } else if (!email.match(regexEmail)) {
      Alert.alert('Error, email tidak valid');
    } else {
      try {
        dispatch(setLoading(true));
        const body = {
          email,
          password,
        };

        const res = await axios.post(`${Login_Api}`, body, {
          validateStatus: status => status < 501,
        });
        console.log(res);
        console.log(res.status);
        if (res.status <= 201) {
          dispatch(setToken(res.data.tokens.access.token));
          navigate('MainRoot');
        } else {
          return Alert.alert('ERROR!!!, wrong username or password');
        }
      } catch (error) {
        console.log(error);
      } finally {
        dispatch(setLoading(false));
      }
    }
  };

  GoogleSignin.configure({
    webClientId:
      '659142653539-pm3v8k6fbo1i3da7oet8i3sqhp3i7v45.apps.googleusercontent.com',
  });

  const signIn = async () => {
    try {
      await GoogleSignin.hasPlayServices();
      const userInfo = await GoogleSignin.signIn();
      console.log(userInfo);
      navigate('MainRoot');
    } catch (error) {
      console.log(error);
      if (error.code === statusCodes.SIGN_IN_CANCELLED) {
        // user cancelled the login flow
      } else if (error.code === statusCodes.IN_PROGRESS) {
        // operation (e.g. sign in) is in progress already
      } else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
        // play services not available or outdated
      } else {
        // some other error happened
      }
    }
  };

  const loginFb = LoginManager.logInWithPermissions(['public_profile']).then(
    function (result) {
      if (result.isCancelled) {
        console.log('Login cancelled');
      } else {
        console.log(
          'Login success with permissions: ' +
            result.grantedPermissions.toString(),
        );
      }
    },
    function (error) {
      console.log('Login fail with error: ' + error);
    },
  );

  const [refresh, setRefresh] = useState(false);
  const onRefreshing = () => {
    setRefresh(true);
    signIn;
    setTimeout(() => {
      setRefresh(false);
    }, 3000);
  };

  const onLogScreenView = async () => {
    try {
      await analytics().logScreenView({
        screen_name: 'Login',
        screen_class: 'Login',
      });
    } catch (error) {}
  };
  /* eslint no-shadow: ["error", { "hoist": "functions" }] */
  /* eslint-env es6 */
  useEffect(() => {
    const tokenCheck = async () => {
      try {
        const res = await AsyncStorage.getItem('@token');
        if (res !== null) {
          dispatch(setToken(res));
          navigation.navigate('MainRoot');
        }
      } catch (error) {
        console.log(error);
      }
    };
    tokenCheck();
    onLogScreenView();
  }, [token, dispatch, navigation]);

  return (
    <View
      refreshControl={
        <RefreshControl refreshing={refresh} onRefresh={onRefreshing} />
      }
      style={styles.wrapper}>
      <StatusBar backgroundColor="white" barStyle="dark-content" />
      <View style={styles.containerInput}>
        <View style={styles.containnerLoginText}>
          <Montserrat style={styles.LoginText}>Login</Montserrat>
        </View>
        <View>
          <TextInput
            style={styles.input}
            placeholder="Enter Email"
            onChangeText={text => setEmail(text)}
          />
        </View>
        <View>
          <TextInput
            style={styles.input}
            secureTextEntry={true}
            placeholder="Enter Password"
            onChangeText={text => setPassword(text)}
          />
        </View>
        <View>
          {loading ? (
            <ActivityIndicator />
          ) : (
            <TouchableOpacity
              onPress={postLogin}
              style={styles.buttonLogin}
              testID="loginButton">
              <Montserrat style={styles.textbuttonLogin}>LOGIN</Montserrat>
            </TouchableOpacity>
          )}
          <Montserrat style={styles.ask}>Don't have an account</Montserrat>
          <TouchableOpacity onPress={() => navigation.navigate('Register')}>
            <Montserrat style={styles.Register}>REGISTER</Montserrat>
          </TouchableOpacity>
          <BioAuth />
          <View
            style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
            <GoogleSigninButton
              style={{width: moderateScale(200), height: moderateScale(50)}}
              size={GoogleSigninButton.Size.Wide}
              color={GoogleSigninButton.Color.Dark}
              onPress={signIn}
              // disabled={this.state.isSigninInProgress}
            />

            <LoginButton
              onPress={loginFb}
              onLoginFinished={(error, result) => {
                if (error) {
                  console.log('login has error: ' + result.error);
                } else if (result.isCancelled) {
                  console.log('login is cancelled.');
                } else {
                  AccessToken.getCurrentAccessToken().then(data => {
                    console.log(data.accessToken.toString());
                    navigate('MainRoot');
                  });
                }
              }}
              onLogoutFinished={() => console.log('logout.')}
            />
          </View>
        </View>
      </View>
    </View>
  );
}

export default Login;
