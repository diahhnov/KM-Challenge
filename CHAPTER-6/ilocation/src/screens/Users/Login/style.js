import {StyleSheet} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import {COLOR} from '../../../assets/color';

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    backgroundColor: COLOR.white,
    justifyContent: 'center',
    alignItems: 'center',
  },
  logo: {
    resizeMode: 'cover',
    justifyContent: 'center',
  },
  containnerLoginText: {
    marginTop: moderateScale(-50),
  },
  LoginText: {
    fontSize: 30,
    fontWeight: 'bold',
    textAlign: 'center',
    color: COLOR.Secondary_Color,
  },
  containerInput: {
    height: moderateScale(300),
    backgroundColor: COLOR.white,
  },
  inSty: {
    width: moderateScale(200),
    marginBottom: moderateScale(18),
  },
  // emailInput: focussed => ({
  //   ...inSty(focussed),
  // }),
  // passwordInput: focussed => ({
  //   ...inSty(focussed),
  // }),
  input: {
    width: moderateScale(200),
    color: COLOR.purple,
  },
  buttonLogin: {
    width: moderateScale(200),
    padding: moderateScale(12),
    marginVertical: moderateScale(15),
    borderRadius: 5,
    backgroundColor: COLOR.Secondary_Color,
  },
  textbuttonLogin: {
    fontWeight: 'bold',
    fontSize: moderateScale(14),
    textAlign: 'center',
    color: COLOR.white,
  },
  ask: {
    textAlign: 'center',
  },
  Register: {
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: moderateScale(14),
    marginTop: moderateScale(3),
  },
});
export default styles;
