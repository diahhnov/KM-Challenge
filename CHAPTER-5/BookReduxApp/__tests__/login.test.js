import React from 'react';
import Login from '../src/screens/Login';
import {create} from 'react-test-renderer';
import Component from '../src/helpers/testComponent';

const component = create(Component(<Login />));
describe('Login', () => {
  test('match a snapshot', async () => {
    expect(component).toMatchSnapshot();
  });
});
