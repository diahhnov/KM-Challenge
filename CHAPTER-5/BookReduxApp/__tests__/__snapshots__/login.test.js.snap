// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login match a snapshot 1`] = `null`;
